### PR TITLE
Change arm64 build to use public VS tools instead of private toolset

### LINKED ISSUE #4823: Build Windows ARM legs using default VS toolset
Similar work was done for coreclr and corefx:
https://github.com/dotnet/corefx/pull/30923
https://github.com/dotnet/coreclr/pull/18817

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes #4823

### CODE REVIEW

**Comment by sdmaclea** [comment]
@rakeshsinghranchi FYI

**Comment by rakeshsinghranchi** [comment]
@sdmaclea , I presume you have confirmed that windows arm64 builds successfully on your devbox. PR build doesn't build Windows Arm64.

**Comment by sdmaclea** [comment]
@dotnet-bot help

**Comment by sdmaclea** [comment]
@rakeshsinghranchi I have created #4837 to try to enable Windows Arm64 CI builds.  Its absence seems to be an oversight left over from when Windows ARM64 SDK was not yet public.

I have not yet tried to build on ,y dev machine...  I will make sure I have the correct dependencies and give it a try.

**Comment by sdmaclea** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -10,7 +10,17 @@ def project = GithubProject
...
 def isPR = true
 
-def platformList = ['Linux:x64:Release', 'Linux:arm:Release', 'Linux:arm64:Release', 'OSX:x64:Release', 'Windows_NT:x64:Release', 'Windows_NT:x86:Debug', 'Windows_NT:arm:Debug', 'Tizen:armel:Release']
+def platformList = [
+  'Linux:x64:Release',
+  'Linux:arm:Release',
+  'Linux:arm64:Release',
+  'OSX:x64:Release',
+  'Windows_NT:x64:Release',
+  'Windows_NT:x86:Debug',
```
Comment (BruceForstall) [inline netci.groovy]: I'm not familiar with this repo, but why are some Debug and some Release?

File: netci.groovy [hunk 2]
```diff
@@ -120,45 +130,6 @@ platformList.each { platform ->
     Utilities.addArchival(newJob, settings)
 }
 
-// **************************
```
Comment (BruceForstall) [inline netci.groovy]: Not related to this change, but I wonder why the Linux arm32 and arm64 docker containers are different from the ones we use in netci.groovy for dotnet/coreclr.

File: src/corehost/build.cmd [hunk 3]
```diff
@@ -126,12 +122,6 @@ exit /b 1
 :GenVSSolution
 :: Regenerate the VS solution
 
-if /i "%__BuildArch%" == "arm64" (
-    REM arm64 builds currently use private toolset which has not been released yet
-    REM TODO, remove once the toolset is open.
-    call :PrivateToolSet
-)
-
```
Comment (BruceForstall) [inline src/corehost/build.cmd]: I think you need to remove the arm64 case below:
```
:: Build the project created by Cmake
if "%__BuildArch%" == "arm64" (
    set __msbuildArgs=/p:UseEnv=true
) else (
    set __msbuildArgs=/p:Platform=%__BuildArch% /p:PlatformToolset="%__PlatformToolset%"
)
```

Comment (sdmaclea) [reply src/corehost/build.cmd]: Thanks missed that...

File: netci.groovy (hunk 1 above)
Comment (sdmaclea) [reply netci.groovy]: I was surprised when I first saw this too.  My hypothesis is that the code in this repo is not very architecture specific.  It is pretty generic C++ code and therefore does not need a lot of per architecture testing.  So it is a bit of a random selection...

**Comment by sdmaclea** [comment]
@dotnet-bot test ci please

**Comment by sdmaclea** [comment]
@dotnet-bot test ci please

File: netci.groovy (hunk 2 above)
Comment (sdmaclea) [reply netci.groovy]: I had to update the arm64 containers to get cross crossgen working last spring.  I don't remember the details, but I might not have updated the coreclr containers to match...

**Comment by sdmaclea** [comment]
The CI machines seem to be using CMake 3.9.3 which doesn't satisfy the cmake 3.10 requirement.

**Comment by sdmaclea** [comment]
Looks like it is using label `win2012-20180911` to select the machines.  Let me look at what coreclr is using.

**Comment by sdmaclea** [comment]
@dotnet-bot test ci please

**Comment by sdmaclea** [comment]
I have manually tested building `win-arm64` on my local dev machine.  The build completed successfully.

I have removed the CI changes, I will move to a separate PR.  I will merge as soon as tests complete.

**Pull request merged**